ads/openx.js

```diff
@@ -0,0 +1,174 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {loadScript, writeScript, checkData} from '../src/3p';
+
+/**
+ * @param {!Window} global
+ * @param {!Object} data
+ */
+export function openx(global, data) {
+  const openxData = ['host', 'nc', 'auid', 'dfpSlot', 'dfp'];
+  const dfpData = Object.assign({}, data);// make a copy for dfp
+  checkData(data, openxData);
+
+  // consolidate doubleclick inputs for forwarding
+  // conversion rules are explained in openx.md
+  if (data.dfpSlot) {
+    // anything starting with 'dfp' gets promoted
+    for (const openxKey of openxData) {
+      if (openxKey in dfpData && openxKey !== 'dfp') {
+        if (openxKey.indexOf('dfp') === 0) {
+          // remove 'dfp' prefix, lowercase first letter
+          let fixKey = openxKey.substring(3);
+          fixKey = fixKey.substring(0,1).toLowerCase() + fixKey.substring(1);
+          dfpData[fixKey] = data[openxKey];
+        }
+        delete dfpData[openxKey];
+      }
+    }
+
+    // promote the whole 'dfp' object
+    if ('dfp' in data) {
+      Object.assign(dfpData, dfpData.dfp);
+      delete dfpData['dfp'];
+    }
+
+    checkData(dfpData, [
+      'slot', 'targeting', 'categoryExclusions',
+      'tagForChildDirectedTreatment', 'cookieOptions',
+      'overrideWidth', 'overrideHeight',
+    ]);
+  }
+
+  let jssdk = `https://${data.host}/mw/1.0/jstag`;
```

Comment:
So, since `host` is required, could you please its existence? You can use the provided `validateDataExists`...